docs/input-validation.rst

```diff
@@ -0,0 +1,129 @@
+Input validation for *Units*
+---------------------------------------------------------------------
+
+:Maintainer: Masatake YAMATO <yamato@redhat.com>
+
+----
+
+We have to maintain parsers for languages that we don't know well.  We
+don't have enough time to learn the languages.
+
+*Units* test cases help us not introduce wrong changes to a parser.
+
+However, there is still an issue; a developer who doesn't know a
+target language well may write a broken test input file for the
+language.  Here comes "Input validation."
+
+You can validate the test input files of *Units* with *validate-input*
+make target if a validator for a language is defined.
+
+How to run and an example session
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is an example validating an input file for JSON.
+
+.. code-block:: console
+
+  $ make validate-input VALIDATORS=jq
+  ...
+  Category: ROOT
+  ------------------------------------------------------------
+  simple-json.d/input.json with jq                                 valid
+
+  Summary
+  ------------------------------------------------------------
+    #valid:                                 1
+    #invalid:                               0
+    #skipped (known invalidation)           0
+    #skipped (validator unavailable)        0
+
+
+This example shows validating *simple-json.d/input.json* as an input
+file with *jq* validator. With VALIDATORS variable passed via
+command-line, You can specify validators to run. Multiple validators
+can be specified using a comma-separated list.  If you don't give
+VALIDATORS, the make target tries to use all available validators.
+
+The meanings of "valid" and "invalid" in "Summary" are apparent.  In
+two cases, the target skips validating input files:
+
+#skipped (known invalidation)
+
+    a test case specifies KNOWN-INVALIDATION in its *validator* file.
+
+#skipped (validator unavailable)
+
+    a backend command for a validator command is not available.
```

Comment:
What do you mean by a "backend command" 
I think you can drop backend and just say. 
`A command for a validator is not available`